comid/rawint.go

```diff
@@ -0,0 +1,330 @@
+		return 0, fmt.Errorf("unexpected type for RawIntInteger: %T", v)
+	}
+}
+
+// TaggedRawIntRangeType is the name of the Range version of RawInt
+const TaggedRawIntRangeType = "rawIntRange"
+
+// TaggedRawIntRange implements the Range version of RawInt. The range is
+// made of minimum and maximum values. If the minimum is nil, it's assumed
+// to be negative infinity. If the maximum is nil, it's assumed to be
+// positive infinity.
+type TaggedRawIntRange struct {
+	Min *int64
+	Max *int64
+}
+
+// NewRawIntRange creates a TaggedRawIntRange with the input value
+func NewRawIntRange(val any) (*TaggedRawIntRange, error) {
+	var ret TaggedRawIntRange
+
+	if val == nil {
+		return &ret, nil
+	}
+
+	switch v := val.(type) {
+	case TaggedRawIntRange:
+		ret = v
+	case *TaggedRawIntRange:
+		ret = *v
+	default:
+		return nil, fmt.Errorf("unexpected type for TaggedRawIntRange: %T", v)
+	}
+
+	return &ret, nil
+}
+
+// Valid checks if TaggedRawIntRange is valid
+func (o TaggedRawIntRange) Valid() error {
+	if o.Min != nil && o.Max != nil && *o.Min > *o.Max {
+		return fmt.Errorf("TaggedRawIntRange: Invalid Range, Min: %d Max: %d", *o.Min, *o.Max)
+	}
+
+	return nil
+}
+
+// String converts TaggedRawIntRange to a string
+func (o TaggedRawIntRange) String() string {
+	var rangeMin, rangeMax string
+
+	if o.Min != nil {
+		rangeMin = fmt.Sprintf("[%d", *o.Min)
+	} else {
+		rangeMin = "(-inf"
+	}
+
+	if o.Max != nil {
+		rangeMax = fmt.Sprintf("%d]", *o.Max)
+	} else {
+		rangeMax = "inf)"
+	}
+
+	return fmt.Sprintf("%s:%s", rangeMin, rangeMax)
+}
+
+// Type returns the name/type of TaggedRawIntRange
+func (o TaggedRawIntRange) Type() string {
+	return TaggedRawIntRangeType
+}
+
+// CompareAgainstRefInteger compares TaggedRawIntRange against a given
+// RawIntInteger reference
+func (o TaggedRawIntRange) CompareAgainstRefInteger(ref RawIntInteger) bool {
+	refVal, err := convertRawIntIntegerToInt64(ref)
+	if err != nil {
+		log.Printf("TaggedRawIntRange:CompareAgainstRefInteger: Error: %v", err)
+		return false
+	}
+
+	if o.Min != nil && refVal == *o.Min && o.Max != nil && refVal == *o.Max {
+		return true
+	}
+
+	return false
+}
+
+// CompareAgainstRefRange compares TaggedRawIntRange against another
+// TaggedRawIntRange reference
+func (o TaggedRawIntRange) CompareAgainstRefRange(ref TaggedRawIntRange) bool {
+	if o.Min != nil && ref.Min != nil && *o.Min < *ref.Min {
+		return false
+	}
+
+	if o.Max != nil && ref.Max != nil && *o.Max > *ref.Max {
+		return false
+	}
+
+	return true
+}
+
+// NewRawIntIntegerType returns a new RawIntInteger from given value
+func NewRawIntIntegerType(val any) (*RawInt, error) {
+	ret, err := NewRawIntInteger(val)
+	if err != nil {
+		return nil, err
+	}
+
+	return &RawInt{ret}, nil
+}
+
+// NewRawIntRangeType returns a new TaggedRawIntRange from given value
+func NewRawIntRangeType(val any) (*RawInt, error) {
+	ret, err := NewRawIntRange(val)
+	if err != nil {
+		return nil, err
+	}
+
+	return &RawInt{ret}, nil
+}
```

Comment:
I don't understand why an int range only takes a single value and returns a non-range.

Reply:
It's a factory function used by `NewRawInt`. The function has a fixed prototype, like below:
`type IRawIntFactory = func(val any) (*RawInt, error)`

That's why it take only one input and returns RawInt (which could either be an Integer or Range based on the function).